### PR TITLE
Add documentation for notifications

### PR DESCRIPTION
**Pull request by ilmiawan**
#Monika PR
## What issue this PR is trying to solve
Currently there is no documentation on how to set the notification configuration and how the notification works in Monika

## How this PR solve the issue
By adding a documentation on how the notification configuration set and how it works

## How to test 
By opening the the documentation and search the notifications

### CODE REVIEW

File: docs/src/pages/guides/notifications.md [hunk 1]
```diff
@@ -1,3 +1,189 @@
-# Notifications
+---
+id: notifications
+title: Notifications
+---
 
-under construction
+## Notification Types
+
+In Monika, currently we provided multiple notifications type. We are implementing email notifications through sendgrid, mailgun, smtp and webhook.
```
Comment (kevinhermawan) [inline docs/src/pages/guides/notifications.md:8]: ~In Monika, currently we provided multiple notifications type.~ Monika provides multiple notification types.

~We are implementing email notifications through sendgrid, mailgun, smtp and webhook.~ You can implement notifications via sendgrid, mailgun, smtp and/or webhook.

File: docs/src/pages/guides/notifications.md [hunk 2]
```diff
@@ -1,3 +1,189 @@
...
+
+## Configurations
+
+Notifications can be send through multiple processes and mail accounts, define through configurations as in the config.json.example.
```
Comment (kevinhermawan) [inline docs/src/pages/guides/notifications.md:12]: You can configure notifications as much as you need with different or same types. To configure notifications you need to define it on `config.json` in `notifications` section.

Is it better?

File: docs/src/pages/guides/notifications.md [hunk 3]
```diff
@@ -1,3 +1,189 @@
...
+```js
+const mg = mailgun({ apiKey: API_KEY, domain: DOMAIN })
+const data = {
+  from: `${sender.name} <${sender.email}>`,
+  to: recipients,
+  subject: subject,
+  text: body,
+}
+return mg.messages().send(data)
+```
```
Comment (raosan) [inline docs/src/pages/guides/notifications.md:84]: I think this part doesn't need to be in the guides. User doesn't need to know our implementation detail, just need to know how to use

File: docs/src/pages/guides/notifications.md [hunk 4]
```diff
@@ -1,3 +1,189 @@
...
+sgMail.setApiKey(API_KEY)
+const msg = {
+  to: recipients,
+  from: sender.email,
+  subject,
+  text: body,
+}
+
+sgMail.send(msg)
+```
```
Comment (raosan) [inline docs/src/pages/guides/notifications.md:113]: I think this part doesn't need to be in the guides. User doesn't need to know our implementation detail, just need to know how to use

File: docs/src/pages/guides/notifications.md [hunk 5]
```diff
@@ -1,3 +1,189 @@
...
+}
+```
+
+```js
+const res = await axios({
+  url: data.url,
+  data: data.body,
+  method: data.method,
+})
+```
```
Comment (raosan) [inline docs/src/pages/guides/notifications.md:189]: I think this part doesn't need to be in the guides. User doesn't need to know our implementation detail, just need to know how to use

File: docs/src/pages/guides/notifications.md [hunk 6]
```diff
@@ -1,3 +1,189 @@
...
+```js
+export const sendSmtpMail = async (transporter: Mail, opt: Mail.Options) => {
+  return transporter.sendMail({
+    from: opt.from,
+    to: opt.to,
+    subject: opt.subject,
+    html: opt.html,
+  })
+}
+```
```
Comment (raosan) [inline docs/src/pages/guides/notifications.md:155]: I think this part doesn't need to be in the guides. User doesn't need to know our implementation detail, just need to know how to use

Comment (ilmiawan) [reply docs/src/pages/guides/notifications.md]: modified

File: docs/src/pages/guides/notifications.md (hunk 4 above)
Comment (ilmiawan) [reply docs/src/pages/guides/notifications.md]: modified

File: docs/src/pages/guides/notifications.md (hunk 3 above)
Comment (ilmiawan) [reply docs/src/pages/guides/notifications.md]: modified

File: docs/src/pages/guides/notifications.md (hunk 2 above)
Comment (ilmiawan) [reply docs/src/pages/guides/notifications.md]: modified

File: docs/src/pages/guides/notifications.md (hunk 1 above)
Comment (ilmiawan) [reply docs/src/pages/guides/notifications.md]: modified

**Review by raosan — APPROVED** [review]

File: docs/src/pages/guides/notifications.md [hunk 7]
```diff
@@ -1,3 +1,214 @@
...
+---
 
-under construction
+## Notification Types
+
+In `Monika`, we provide multiple notification channels. Notifications are available through sendgrid, mailgun, smtp and webhook.
+
+## Configurations
+
+Notifications can be sent through multiple channels and mail accounts, each are defined through configuration such as in the file `config.json.example` below. All you need for having a notification is set these configurations.
```
Comment (nicnocquee) [inline docs/src/pages/guides/notifications.md:12]: You can remove this part: "All you need for having a notification is set these configurations."

**Review by newm4n — APPROVED** [review]
LGTM

File: docs/src/pages/guides/notifications.md [hunk 8]
```diff
@@ -1,3 +1,214 @@
...
+        "method": "POST",
+        "url": "https://examplewebhookurl.com/webhook"
+      }
+    }
+  ],
+```
+
+## Mailgun
+
+Mailgun is the notification channel through the mailgun email service. For this you need to have a mailgun account first, using your mailgun's API_KEY and DOMAIN you can send notifications to the recipient's mail.
```
Comment (nicnocquee) [inline docs/src/pages/guides/notifications.md:59]: Mailgun is the notification channel through the mailgun email service. First, you need to have a mailgun account, then get the API key and DOMAIN. Please check out the [documentation in mailgun](https://documentation.mailgun.com/en/latest/quickstart.html). Put the API key and the DOMAIN in the monika's configuration as follows.

File: docs/src/pages/guides/notifications.md [hunk 9]
```diff
@@ -1,3 +1,214 @@
...
+Notification ID, to identify the notification number.
+
+**Type**
+Type of Notification in this case it would be set as `mailgun`.
+
+**Data**
+The detail settings needed to send a mailgun.
+
+**Recipients**
+Email addresses that will accept the mail sent from your mailgun account, the recipients address is set in array making it able to send to multiple addresses. `["monika@testmail.com", "symon@testmail.com"]`
```
Comment (nicnocquee) [inline docs/src/pages/guides/notifications.md:83]: An array of email addresses that will receive the e-mail from Monika.

File: docs/src/pages/guides/notifications.md [hunk 10]
```diff
@@ -1,3 +1,214 @@
...
+
+**Api Key**
+Mailgun's account api key, mailgun's registered key to identify your account. ex. `70e34aba-0ea908325`
+
+**Domain**
+A domain that you have set in mailgun. ex. `sandboxmail.mailgun.com`
+
+`Monika` filters the notification types for mailgun and breaks down the data configuration, setting up a mailgun service and then sending through the data using mailgun-js lib.
```
Comment (nicnocquee) [inline docs/src/pages/guides/notifications.md:91]: I think this part can be removed.

File: docs/src/pages/guides/notifications.md [hunk 11]
```diff
@@ -1,3 +1,214 @@
...
+
+## Sendgrid
+
+Similar to mailgun, sendgrid is also an email delivery service. You need to have a sendgrid account before you can send an email through this method. For sending a notification through sendgrid all you need to set is the API_KEY and recipient email addresses.
```
Comment (nicnocquee) [inline docs/src/pages/guides/notifications.md:95]: Similar to mailgun, sendgrid is also an email delivery service. First, sign up for an account in sendgrid and get the API key. Please check [sendgrid's documentation](https://sendgrid.com/docs/api-reference/). Then put the API key in Monika's configuration as follows.

File: docs/src/pages/guides/notifications.md [hunk 12]
```diff
@@ -1,3 +1,214 @@
...
+  "type": "sendgrid",
+  "data": {
+    "recipients": ["mantap@jiwa.com"],
+    "apiKey": "YOUR_API_KEY"
+  }
+}
+```
+
+**ID**
+Notification ID, to identify the notification number.
```
Comment (nicnocquee) [inline docs/src/pages/guides/notifications.md:109]: to identify this notification.

File: docs/src/pages/guides/notifications.md [hunk 13]
```diff
@@ -1,3 +1,214 @@
...
+
+**Type**
+Type of Notification in this case it would be set as `sendgrid`.
+
+**Data**
+The detail settings needed for sending email through sendgrid.
+
+**Recipients**
+Recipient email addresses, set as an array for sending to multiple addresses. ex. `["monika@testmail.com", "symon@testmail.com"]`
```
Comment (nicnocquee) [inline docs/src/pages/guides/notifications.md:118]: An array of email addresses that will receive the notifications from Monika.

File: docs/src/pages/guides/notifications.md [hunk 14]
```diff
@@ -1,3 +1,214 @@
...
+
+**Api Key**
+The api key that sendgrid gives for your account.
+
+`Monika` is using @sendgrid/mail library for sending the sendgrid type notification. `Monika` will breakdown the configuration set the sengrid type API_KEY settings and send the message using the library.
```
Comment (nicnocquee) [inline docs/src/pages/guides/notifications.md:123]: No need to mention this.

File: docs/src/pages/guides/notifications.md [hunk 15]
```diff
@@ -1,3 +1,214 @@
...
+
+## SMTP
+
+SMTP (Simple Mail Transfer Protocol) is an email service using the TCP/IP protocol. For this you need to have an active email account to be used as transporter, a middle service that relays the email sending process.
```
Comment (nicnocquee) [inline docs/src/pages/guides/notifications.md:127]: SMTP (Simple Mail Transfer Protocol) is a way to send emails using the TCP/IP protocol. You can use your own SMTP server or third party's SMTP servers such as Gmail, Fastmail, etc. Then put the hostname, port, username, and password of the SMTP server in Monika's configuration as follows.

File: docs/src/pages/guides/notifications.md [hunk 16]
```diff
@@ -1,3 +1,214 @@
...
+    "hostname": "smtp.mail.com",
+    "port": 222,
+    "username": "dennypradipta",
+    "password": "bismillah"
+  }
+}
+```
+
+**ID**
+Notification ID, to identify the notification number.
```
Comment (nicnocquee) [inline docs/src/pages/guides/notifications.md:145]: to identify this notification.

File: docs/src/pages/guides/notifications.md [hunk 17]
```diff
@@ -1,3 +1,214 @@
...
+
+**Type**
+Type of Notification in this case it would be set as `smtp`.
+
+**Data**
+The detail settings needed for sending email through smtp.
+
+**Recipients**
+Recipient email addresses, set as an array for sending to multiple addresses. ex. `["monika@testmail.com", "symon@testmail.com"]`
```
Comment (nicnocquee) [inline docs/src/pages/guides/notifications.md:154]: An array of email addresses that will receive the notifications from Monika.

File: docs/src/pages/guides/notifications.md [hunk 18]
```diff
@@ -1,3 +1,214 @@
...
+**Port**
+The port that has been allowed to be used for sending mail in your host. ex. `587`
+
+**Username**
+The username that has been registered on your smtp server. ex. `yourusername@gmail.com`
+
+**Password**
+The password set for your username. ex. `thepasswordforyourusername`
+
+`Monika` will breakdown the configuration with type `smtp`, registering the username and password to the mail host with certain port and create a transporter to send to your recipient addresses.
```
Comment (nicnocquee) [inline docs/src/pages/guides/notifications.md:168]: You can remove this part.

File: docs/src/pages/guides/notifications.md [hunk 19]
```diff
@@ -1,3 +1,214 @@
...
+
+## Webhook
+
+Besides the email notifications, `Monika` also provide webhook service using REST API calls and POST methods.
```
Comment (nicnocquee) [inline docs/src/pages/guides/notifications.md:172]: Besides the email notifications, Monika can also send the alerts via web hook. You need to define the URL and the HTTP method in Monika's configuration as follows.

File: docs/src/pages/guides/notifications.md [hunk 20]
```diff
@@ -1,3 +1,214 @@
...
+  "data": {
+    "recipients": ["mantap@jiwa.com"],
+    "method": "POST",
+    "url": "https://examplewebhookurl.com/webhook"
+  }
+}
+```
+
+**ID**
+Notification ID, to identify the notification number.
```
Comment (nicnocquee) [inline docs/src/pages/guides/notifications.md:187]: to identify this notification.

File: docs/src/pages/guides/notifications.md [hunk 21]
```diff
@@ -1,3 +1,214 @@
...
+
+**Type**
+Type of Notification in this case it would be set as `webhook`.
+
+**Data**
+The detail settings needed for sending email through smtp.
```
Comment (nicnocquee) [inline docs/src/pages/guides/notifications.md:193]: for sending alerts through webhook.

File: docs/src/pages/guides/notifications.md [hunk 22]
```diff
@@ -1,3 +1,214 @@
...
+
+**Recipients**
+Recipient email addresses, set as an array for sending to multiple addresses. ex. `["monika@testmail.com", "symon@testmail.com"]`
```
Comment (nicnocquee) [inline docs/src/pages/guides/notifications.md:196]: An array of email addresses that will receive the notifications from Monika.

File: docs/src/pages/guides/notifications.md [hunk 23]
```diff
@@ -1,3 +1,214 @@
...
+
+**Method**
+The method will be used in webhook. ex. `POST`
```
Comment (nicnocquee) [inline docs/src/pages/guides/notifications.md:199]: A valid HTTP method.

File: docs/src/pages/guides/notifications.md [hunk 24]
```diff
@@ -1,3 +1,214 @@
...
+
+**Url**
+The url that you will send notifications to. ex. `https://yourwebsite.com/webhook`
```
Comment (nicnocquee) [inline docs/src/pages/guides/notifications.md:202]: The URL of the server that will receive the webhook notification.

File: docs/src/pages/guides/notifications.md [hunk 25]
```diff
@@ -1,3 +1,214 @@
...
+
+Using the webhook type configuration, `Monika` will send a post request and data through its body. The data for notification will be sent as set in this object :
```
Comment (nicnocquee) [inline docs/src/pages/guides/notifications.md:204]: ... Monika will send a request with the following body:

**Review by nicnocquee — CHANGES_REQUESTED** [review]

**Pull request merged**